vector/src/main/java/im/vector/app/features/auth/ReAuthActivity.kt

```diff
@@ -0,0 +1,225 @@
+            val session: String?,
+            val lastErrorCode: String?,
+            val resultKeyStoreAlias: String
+    ) : Parcelable
+
+    // For sso
+    private var customTabsServiceConnection: CustomTabsServiceConnection? = null
+    private var customTabsClient: CustomTabsClient? = null
+    private var customTabsSession: CustomTabsSession? = null
+
+    @Inject lateinit var authenticationService: AuthenticationService
+    @Inject lateinit var reAuthViewModelFactory: ReAuthViewModel.Factory
+
+    override fun create(initialState: ReAuthState) = reAuthViewModelFactory.create(initialState)
+
+    override fun injectWith(injector: ScreenComponent) {
+        super.injectWith(injector)
+        injector.inject(this)
+    }
+
+    private val sharedViewModel: ReAuthViewModel by viewModel()
+
+    // override fun getTitleRes() = R.string.re_authentication_activity_title
+
+    override fun initUiAndData() {
+        super.initUiAndData()
+
+        val title = intent.extras?.getString(EXTRA_REASON_TITLE) ?: getString(R.string.re_authentication_activity_title)
+        supportActionBar?.setTitle(title) ?: run { setTitle(title) }
+
+//        val authArgs = intent.getParcelableExtra<Args>(MvRx.KEY_ARG)
+
+        // For the sso flow we can for now only rely on the fallback flow, that handles all
+        // the UI, due to the sandbox nature of CCT (chrome custom tab) we cannot get much information
+        // on how the process did go :/
+        // so we assume that after the user close the tab we return success and let caller retry the UIA flow :/
+
+        addFragment(
+                R.id.container,
+                PromptFragment::class.java
+        )
+
+        sharedViewModel.observeViewEvents {
+            when (it) {
+                is ReAuthEvents.OpenSsoURl -> {
+                    openInCustomTab(it.url)
+                }
+                ReAuthEvents.Dismiss -> {
+                    setResult(RESULT_CANCELED)
+                    finish()
+                }
+                is ReAuthEvents.PasswordFinishSuccess -> {
+                    setResult(RESULT_OK, Intent().apply {
+                        putExtra(RESULT_FLOW_TYPE, LoginFlowTypes.PASSWORD)
+                        putExtra(RESULT_VALUE, it.passwordSafeForIntent)
+                    })
+                    finish()
+                }
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        // It's the only way we have to know if sso falback flow was successful
+        withState(sharedViewModel) {
+            if (it.ssoFallbackPageWasShown) {
+                Timber.d("## UIA ssoFallbackPageWasShown tentative success")
+                setResult(RESULT_OK, Intent().apply {
+                    putExtra(RESULT_FLOW_TYPE, LoginFlowTypes.SSO)
+                })
+                finish()
+            }
+        }
+    }
+
+    override fun onStart() = withState(sharedViewModel) { state ->
+        super.onStart()
```

Comment:
weird to call `super` inside the `withState` block.

Reply:
+1